src/blog/konfig-manager/intro-spring-boot.md

```diff
@@ -0,0 +1,352 @@
+---
+Author: Saul Nachman & Moshe Immerman
+Last updated: 23/07/2021
+---
+
+# Konfig Manager
+
+<!-- markdownlint-disable MD036 -->
+**Managing a Spring Boot configuration**
+<!-- markdownlint-enable MD036 -->
+
+The Konfig Manager is for platform engineers needing to manage overlapping configurations across applications and environments.
+
+## Design principles
+
+- **100% YAML** – `Konfigs` are valid YAML and IDE validation and autocomplete of k8s resources works as normal.
+- **Simple** – Easy to use and quick to get started.
+- **Reconciliation based** – Changes are applied quickly and resiliently (unlike webhooks) at runtime.
+
+## Installation
+
+### Prerequisites
+
+This guide assumes you have either a [kind cluster](https://kind.sigs.k8s.io/docs/user/quick-start/) or [minikube cluster](https://minikube.sigs.k8s.io/docs/start/) running, or have some other way of interacting with a cluster via [kubectl](https://kubernetes.io/docs/tasks/tools/).
+
+To test the deployment in browser, set up ingress to your cluster – see, for example, the [minikube](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/#create-an-ingress-resource) or [kind](https://kind.sigs.k8s.io/docs/user/ingress) docs.
+
+### Install
+
+```bash
+export VERSION=0.10.1
+# For the latest release version: https://github.com/flanksource/konfig-manager/releases
+
+# Apply the operator
+kubectl apply -f https://github.com/flanksource/konfig-manager/releases/download/v${VERSION}/operator.yml
+```
+
+### Following the logs
+
+To follow the konfig-manager logs, open a new terminal and, changing what needs to be changed, run:
+
+```bash
+kubectl logs -f --since 10m -n konfig-manager konfig-manager-d54f64588-hjq5z konfig-manager
+```
+
+These logs are where reconciliation successes and errors show up – and the best place to look when debugging.
+
+## Use case: Externalising a Spring Boot configuration
+
+*As a platform engineer, I need to manage overlapping configurations across multiple applications and environments, so that applications are easy to promote.*
+
+The Java Spring Boot project [recommends developers externalise their configuration](https://docs.spring.io/spring-boot/docs/1.0.1.RELEASE/reference/html/boot-features-external-config.html) and one method it provides is the `application.properties` file.
+
+The Konfig Manager makes it easy to convert a set of `ConfigMaps` and `Secrets` to a single output  – the `application.properties` file – providing a single configuration to the container file system.
+
+The `Konfig` type allows developers to specify a `hierarchy` (defining merge inclusion and precedence) and an `output` field (defining output structure).
+
+### Step 1: Apply configmaps, namespace and secrets
+
+Before you apply the `Konfig`, ensure that there are config objects in place (for the `hierarchy`, see [Step 2](#Step-2-Apply-a-konfig)):
+
+```yaml
+cat <<'EOF' | kubectl apply -f -
```

Comment:
```suggestion
cat <<EOF | kubectl apply -f -
```

Reply:
Okay... not sure if I understand what exactly that means.... can you give an example or a link that explains this?
And I see we only have this single quoting on this EOF and not on any other... is this intentional?

Reply:
It's intentional, yeah. It's because this one contains ${VALUE}.

Not 100% sure why it works. Didn't go down the rabbit hole.